### PR TITLE
New package: add Typst 0.1.0

### PR DESCRIPTION
**Pull request by lucifer1004**

### CODE REVIEW

File: T/Typst/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+sources = [
+    GitSource("https://github.com/typst/typst.git", "b3faef4b80a674294091066e20501e3a5d0f6103")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/typst
+cargo build -p typst-cli --release
+mkdir -p $bindir
+mv target/$rust_target/release/typst $bindir/
```
Comment (giordano) [inline T/Typst/build_tarballs.jl:18]:
```suggestion
install -Dvm 755 "target/${rust_target}/release/typst" "${bindir}/typst${exeext}"
```

File: T/Typst/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+install_license LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = filter(!Sys.iswindows, supported_platforms())
```
Comment (giordano) [inline T/Typst/build_tarballs.jl:24]: What's the problem with 64-bit windows?

Comment (lucifer1004) [reply T/Typst/build_tarballs.jl:23]: See https://buildkite.com/julialang/yggdrasil/builds/2119#0187561d-fa2f-458c-b0a5-50eb1791a347

**Comment by lucifer1004** [comment]
@giordano Can this be merged?

File: T/Typst/build_tarballs.jl (hunk 2 above)
Comment (lucifer1004) [reply T/Typst/build_tarballs.jl:23]: Yes, we definitely need to improve this.

Comment (giordano) [reply T/Typst/build_tarballs.jl:23]: > This is probably not your fault, it’s probably a bug in BB.

I think this _is_ @lucifer1004's fault, and _not_ a bug in BB:
```console
sandbox:${WORKSPACE}/srcdir/typst # ls target/${rust_target}/release/typst*
target/x86_64-pc-windows-gnu/release/typst.d    target/x86_64-pc-windows-gnu/release/typst.exe
```
The file to install should have been `target/${rust_target}/release/typst${exeext}`, not `target/${rust_target}/release/typst`

**Pull request merged**